core/services/pipeline/orm.go

```diff
@@ -78,71 +75,15 @@ func NewORM(db *gorm.DB, config Config, eventBroadcaster postgres.EventBroadcast
 
 // The tx argument must be an already started transaction.
 func (o *orm) CreateSpec(ctx context.Context, tx *gorm.DB, taskDAG TaskDAG, maxTaskDuration models.Interval) (int32, error) {
-	var specID int32
 	spec := Spec{
 		DotDagSource:    taskDAG.DOTSource,
 		MaxTaskDuration: maxTaskDuration,
 	}
 	err := tx.Create(&spec).Error
 	if err != nil {
-		return specID, err
-	}
-	specID = spec.ID
-
-	// Create the pipeline task specs in dependency order so
-	// that we know what the successor ID for each task is
-	tasks, err := taskDAG.TasksInDependencyOrder()
-	if err != nil {
-		return specID, err
-	}
-
-	// Create the final result task that collects the answers from the pipeline's
-	// outputs.  This is a Postgres-related performance optimization.
-	resultTask := ResultTask{BaseTask{dotID: ResultTaskDotID}}
-	for _, task := range tasks {
-		if task.DotID() == ResultTaskDotID {
-			return specID, errors.Errorf("%v is a reserved keyword and cannot be used in job specs", ResultTaskDotID)
-		}
-		if task.OutputTask() == nil {
-			task.SetOutputTask(&resultTask)
-		}
+		return 0, err
 	}
-	tasks = append([]Task{&resultTask}, tasks...)
-
-	taskSpecIDs := make(map[Task]int32)
```

Comment:
win #1 - we don't have to insert the tasks one by one to populate their successor IDs